### PR TITLE
Update the module to work with InvenioRDM v9+

### PR DESCRIPTION
**Pull request by max-moser**
Modernize the module in a way that it works with InvenioRDM v9+

* closes https://github.com/inveniosoftware/invenio-stats/issues/119

---

Edit: The changes made in this PR:
```
Version 2.0.0 (WIP)
-------------

- Upgrade to ``invenio-search`` 2.x
- Drop support for Elasticsearch 2, 5, and 6
- Add support for OpenSearch 1 and 2
- Drop support for Python 2.7 and 3.6
- Remove function ``invenio_stats.utils:get_doctype``
- Update dependencies
```

### CODE REVIEW

**Comment by max-moser** [comment]
- [ ] The documentation in `__init__.py` still needs to be updated
- [ ] `doc_type`s are still referenced in `tests/test_cli.py` (e.g. [here](https://github.com/max-moser/invenio-stats/blob/mm/update/tests/test_cli.py#L139)), but that doesn't seem to cause the tests to fail - what should we do here?

File: CHANGES.rst [hunk 1]
```diff
@@ -9,6 +9,16 @@
 Changes
 =======
 
+Version 2.0.0
```
Comment (kpsherva) [inline CHANGES.rst:12]: do we need a release already ? or you kept it as a changelog?

Comment (max-moser) [reply CHANGES.rst:12]: Mostly as a changelog – until yesterday, the commit was still prefixed with "WIP"... Alternatively, we could drop that commit and track the changes somewhere in an issue.

File: examples/app.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 #
 # This file is part of Invenio.
 # Copyright (C) 2016-2018 CERN.
+# Copyright (C)      2022 TU Wien.
```
Comment (ntarocco) [inline examples/app.py:5]: The example app can be completely removed, we are removing them.

File: invenio_stats/views.py [hunk 3]
```diff
@@ -49,37 +49,52 @@ def post(self, **kwargs):
 
         result = {}
         for query_name, config in data.items():
-            if config is None or not isinstance(config, dict) \
-                    or (set(config.keys()) != {'stat', 'params'} and
-                        set(config.keys()) != {'stat'}):
+            if (
```
Comment (ntarocco) [inline invenio_stats/views.py:52]: This `if` is so complex that even black has trouble to format the code! 👀 🤣

**Review by ntarocco — APPROVED** [review]
Thanks a lot for this refactor!

File: examples/app.py (hunk 2 above)
Comment (max-moser) [reply examples/app.py:5]: Created an issue for this: https://github.com/inveniosoftware/invenio-stats/issues/123

**Comment by max-moser** [comment]
i interpret the silence as no veto against the slight documentation changes (which mostly consisted of updating mentions of ES, and a bit of example code reformatting) :stuck_out_tongue: 
and if something is undesired, we can always still create a follow-up PR

**Pull request merged**